client/state/comments/reducer.js

```diff
@@ -78,21 +77,11 @@ export function items( state = {}, action ) {
 				[ stateKey ]: reject( state[ stateKey ], { ID: commentId } )
 			};
 		case COMMENTS_LIKE:
-			const { like_count } = action;
-			return {
-				...state,
-				[ stateKey ]: map( state[ stateKey ], updateComment( commentId, { i_like: true, like_count } ) )
-			};
-		case COMMENTS_LIKE_UPDATE:
-			const { iLike, likeCount } = action;
-			return {
-				...state,
-				[ stateKey ]: map( state[ stateKey ], updateComment( commentId, { i_like: iLike, like_count: likeCount } ) )
-			};
 		case COMMENTS_UNLIKE:
+			const { like_count } = action;
 			return {
 				...state,
-				[ stateKey ]: map( state[ stateKey ], updateComment( commentId, { i_like: false } ) )
+				[ stateKey ]: map( state[ stateKey ], updateComment( commentId, { i_like: type === COMMENTS_LIKE, like_count } ) )
```

Comment:
I see the value in the fall-through logic here, but we generally like to avoid it. if it's only saving us a line or two we might consider just leaving the two branches separate. the logic on `i_like`, for example, is a bit confusing to me when I see it right under `COMMENTS_UNLIKE`